scripts/modules/b2b-account/quotes.js

```diff
@@ -369,14 +381,60 @@ define([
                 window.location = '/myaccount/quote/' + quoteId + '/edit';
             }
         },
-        deleteQuote: function () {
-            this.trigger('deleteQuoteView');
+        deleteQuote: function (e, row) {
+            e.stopPropagation();
+            var self=this;
+           var confirmFlag= window.confirm("Do you want to Delete " +row.attributes.name+"?");
+           if(confirmFlag){
+            var quote = new QuoteModels.Quote({
+                id:row.id
+            });
+            return quote.apiModel.removeQuote().then(function (res) {
+                self.refreshGrid();
+            }, function (error) {
+              self.showMessageBar(error.message);
+            });}
         },
-        copyQuote: function () {
-            this.set('copyQuoteView');
+        copyQuote: function (e, row) {
+            var self=this;
+            e.stopPropagation();
+            var quote = new QuoteModels.Quote({
+               id:row.id
+            });
+            return quote.apiModel.copy().then(function (res) {
+                if (isSalesRep) {
+                    window.location = '/selleraccount/quote/' + res.data.id + '/edit';
+                } else {
+                    window.location = '/myaccount/quote/' + res.data.id + '/edit';
+                }
+            }, function (error) {
+                self.showMessageBar(error.message);
+            });
+        },
+        emailQuote: function (e, row) {
+            e.stopPropagation();
+            emailQuoteView.renderView();
+            emailQuoteView.render(row.id);
         },
-        emailQuote: function () {
-            this.set('emailQuoteView');
+        viewQuote: function (e, row) {
+            if (isSalesRep) {
+                window.location = '/selleraccount/quote/' + row.id;
+            }
+            else {
+                window.location = '/myaccount/quote/' + row.id;
+            }
+        },
+        showMessageBar: function (error) {
+            var self = this;
```

Comment:
i think we have to put showMessageBar() method in some common place and use it.

Reply:
Use case is different here. If I render self then it goes back to its default state and clears the input field.